OKTA-473379 Add LINE social IdP guide (#3125)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/social-login/main/line/noemail.md

## Handle users without email addresses

> 

Reviewer: Can we add a blurb about what to do if you can't see a user's email address?

It's my understanding that LINE requires you to do something in order to collect email addresses. See here: https://developers.line.biz/en/docs/line-login/integrate-line-login/#applying-for-email-permission. So my thought is to have something like

> ###

```suggestion
### The user has an email address in LINE, but it's not showing up in Okta
Before you proceed to the next section, test your integration with a user who has an email in LINE. If you don't see their email in Okta, check that you have [email address permission](https://developers.line.biz/en/docs/line-login/integrate-line-login/#applying-for-email-permission) in the channel you are using. If you have confirmed that you have this permission and are still not able to see their emails in Okta, contact support.

### The user doesn't have an email address in LINE
```

## Handle users without email addresses

### The user has an email address in LINE, but it's not in Okta

> Before you proceed to the next section, test your integration with a user who has an email in LINE. If you don't see their email in Okta, verify that you have [email address permission](https://developers.line.biz/en/docs/line-login/integrate-line-login/#applying-for-email-permission) in your LINE channel. If you've confirmed that your channel has email permission and you're still unable to see user emails in Okta, [contact support](https://support.okta.com/help/s/opencase).

Reviewer 2: The link text reads awkward to me. I know it's the content in the URL, but I wonder what "email address permission" is. The header content in the Link docs is more clear, imo.
```suggestion
Before you proceed to the next section, test your integration with a user who has an email in LINE. If you don't see their email in Okta, verify that you have permission to [access the user's email address](https://developers.line.biz/en/docs/line-login/integrate-line-login/#applying-for-email-permission) in your LINE channel. If you've confirmed that your channel has email permission and you're still unable to see user emails in Okta, [contact support](https://support.okta.com/help/s/opencase).
```

### The user doesn't have an email address in LINE

LINE doesn’t always provide email addresses for users that it authenticates, but Okta requires an email address for its users to be able to sign in. You can support users who don't have email addresses by using information from LINE to generate email addresses for them.

For example, you might generate email addresses using the `example.com` domain. This ensures that your LINE users have email addresses that Okta can use, but can be easily identified as invalid should you ever want to replace them with valid email addresses. They're guaranteed to be invalid as `example.com` is a reserved domain. See [Reserved Top Level DNS Names (RFC2606)](https://datatracker.ietf.org/doc/html/rfc2606) for more information on reserved domains.

You can customize this mapping by using the [Okta Expression Language](/docs/reference/okta-expression-language/). See [Profile Editor](https://help.okta.com/okta_help.htm?id=ext_app_map) for more information on attribute mapping.

> To generate user login values and email addresses for LINE users, do the following:

Reviewer 3: ```suggestion
To generate user sign-in values and email addresses for LINE users, do the following:
```

1. From the Admin Console, click **Security**.

1. Click **Identity Providers**.

> 1. Locate LINE in the list of providers, then click **Configure** > **Edit Profile and Mappings**.

Reviewer 3: ```suggestion
1. Locate LINE in the list of providers, and then click **Configure** > **Edit Profile and Mappings**.
```

1. Click **Mappings**.

1. Select the tab that maps a LINE user to an Okta user.

1. Set the expression that maps to the Okta user's `login` value to:
   ```

>    appuser.email == null ? appuser.id + '@line.example.com' : appuser.email

Reviewer: This needs to be `appuser.sub`.

   ```

1. Set the expression that maps to the Okta user's `email` value to:
   ```

>    appuser.email == null ? appuser.id + '@line.example.com' : appuser.email

Reviewer: This needs to be `appuser.sub`.

   ```

1. Click **Save Mappings**

Similarly, you can map the IdP username by doing the following:

1. From the Admin Console, click **Security**.

1. Click **Identity Providers**.

1. Locate LINE in the list of providers, and then click **Configure** > **Configure Identity Provider**.

1. Set the expression for **IdP Username** to: 
```

>    idpuser.email == null ? idpuser.id + '@line.example.com' : idpuser.email

Reviewer: This needs to be `idpuser.sub`.
